Automattic/sensei
# Deprecate experimental rest api

### Changes proposed in this Pull Request

* Deprecates experimental Sensei REST API. The endpoints that are deprecated are:
    * courses
    * modules
    * version
* The endpoints were always behind a feature flag so it is expected that very few users are affected by it.
* Apart from the deprecation warnings, all endpoints will now return a warning header `Warning: 299 - Deprecated API`
* The data model code which was only used by the API was deprecated as well.


### Testing instructions

* Do a smoke test and observe that no warnings are logged.
* Optionally enable the feature by using `define( 'SENSEI_FEATURE_FLAG_REST_API_V1', true );`. Try out the endpoints and observe that warnings are created and the mentioned header is returned.
